app/controllers/application_controller.rb

```diff
@@ -60,4 +60,15 @@ def basecamp_auth_check
       @basecamp_agent = current_user.agents.where(type: 'Agents::BasecampAgent').first
     end
   end
+
+  def agent_params
+    return {} unless params[:agent]
+    @agent_params ||= begin
+      options = params[:agent].delete(:options) if params[:agent][:options].present?
```

Comment:
I don't think you need the `if` because `params[:agent].delete(:options)` will be `nil` if the options param isn't there, right?